File: tests/claude-plugin-integration/test-repos/ruby-sample/payment_service.rb
require_relative 'user'

class PaymentService
  def process_payment(user, amount)
    puts "Processing $#{amount} for #{user.name}"
    { success: true, amount: amount, user: user.name }
  end

  def refund_payment(user, amount)
    puts "Refunding $#{amount} to #{user.name}"
    { success: true, amount: amount }
  end
end

File: tests/claude-plugin-integration/test-repos/python-sample/payment_service.py
from user import User

class PaymentService:
    def process_payment(self, user: User, amount: float) -> dict:
        print(f'Processing ${amount} for {user.name}')
        return {'success': True, 'amount': amount, 'user': user.name}

    def refund_payment(self, user: User, amount: float) -> dict:
        print(f'Refunding ${amount} to {user.name}')
        return {'success': True, 'amount': amount}
